File: app/src/main/java/org/neuverse/entity/Empresa.java
package org.neuverse.entity;
import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;

/**
 * Entity representing a row in the {@code core.empresas} table.  Using Panache
 * simplifies CRUD operations, letting us write endpoints that call
 * {@link Empresa#listAll()} or {@link Empresa#findById(Object)} directly.
 */
@Entity
@Table(name = "empresas", schema = "core")
public class Empresa extends PanacheEntityBase {
    @Id
    @Column(name = "id", nullable = false)
    public UUID id;

    @Column(name = "nome", nullable = false)
    public String nome;

    @Column(name = "slug", nullable = false)
    public String slug;

    @Column(name = "cnpj")
    public String cnpj;

    @Column(name = "status", nullable = false)
    public String status;

    @Column(name = "criado_em", nullable = false)
    public OffsetDateTime criado_em;

    @Column(name = "atualizado_em", nullable = false)
    public OffsetDateTime atualizado_em;
}

File: back-end/src/main/kotlin/org/neuverse/entity/Empresa.kt
package org.neuverse.entity

import jakarta.persistence.*
import java.time.OffsetDateTime
import java.time.ZoneOffset
import java.util.UUID

@Entity
@Table(name = "empresas", schema = "core")
class Empresa(

    @Id
    @Column(name = "id", nullable = false, updatable = false)
    var id: UUID = UUID.randomUUID(),

    @Column(name = "nome", nullable = false)
    var nome: String = "",

    @Column(name = "criado_em", nullable = false)
    var criadoEm: OffsetDateTime = OffsetDateTime.now(ZoneOffset.UTC),

    /**
     * Empresa pode ter 1..N diretores/donos.
     * Usa uma tabela de junção core.empresa_diretores
     *  - empresa_id (UUID)
     *  - diretor_id (UUID -> usuarios.id)
     */
    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "empresa_diretores",
        schema = "core",
        joinColumns = [JoinColumn(name = "empresa_id")],
        inverseJoinColumns = [JoinColumn(name = "diretor_id")]
    )
    var diretores: MutableSet<Usuario> = mutableSetOf()
)
